tests/e2e-pw/specs/shopper/shopper-checkout-save-card-and-purchase.spec.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import test, { Page, expect } from '@playwright/test';
+
+/**
+ * Internal dependencies
+ */
+import { config } from '../../config/default';
+import { getAnonymousShopper } from '../../utils/helpers';
+import {
+	confirmCardAuthentication,
+	deleteSavedCard,
+	fillCardDetails,
+	placeOrder,
+	placeOrderWithOptions,
+	selectSavedCardOnCheckout,
+	setSavePaymentMethod,
+	setupProductCheckout,
+} from '../../utils/shopper';
+import { goToMyAccount, goToShop } from '../../utils/shopper-navigation';
+import RestAPI from '../../utils/rest-api';
+
+type CardType = [ string, typeof config.cards.basic ];
+
+const cards: Array< CardType > = [
+	[ 'basic', config.cards.basic ],
+	[ '3ds', config.cards[ '3ds' ] ],
+];
+
+test.describe( 'Saved cards', () => {
+	cards.forEach( ( [ cardType, card ] ) => {
+		test.describe(
+			`When using a ${ cardType } card added on checkout`,
+			() => {
+				let shopperPage: Page;
+				test.beforeAll( async ( { browser }, { project } ) => {
+					const restApi = new RestAPI( project.use.baseURL );
+					restApi.deleteCustomerByEmailAddress(
+						config.addresses.customer.billing.email
+					);
+					shopperPage = ( await getAnonymousShopper( browser ) )
+						.shopperPage;
+					await placeOrderWithOptions( shopperPage, {
+						billingAddress: config.addresses.customer.billing,
+						createAccount: true,
+					} );
+				} );
+				test( 'should save the card', async ( {} ) => {
+					await goToShop( shopperPage );
+					await setupProductCheckout( shopperPage );
+					await fillCardDetails( shopperPage, card );
+					await setSavePaymentMethod( shopperPage, true );
+					await placeOrder( shopperPage );
+					if ( cardType === '3ds' ) {
+						await confirmCardAuthentication( shopperPage );
+					}
+					await expect(
+						shopperPage.getByText( 'Order received' ).first()
+					).toBeVisible();
+
+					await goToMyAccount( shopperPage, 'payment-methods' );
+					await expect(
+						shopperPage.getByText( card.label )
+					).toBeVisible();
+					await expect(
+						shopperPage.getByText(
+							`${ card.expires.month }/${ card.expires.year }`
+						)
+					).toBeVisible();
+				} );
+
+				test( 'should process a pament with the saved card', async ( {} ) => {
```

Comment:
Nitpick: typo. Payment is missing the y.

Reply:
Fixed in [7f62120](https://github.com/Automattic/woocommerce-payments/pull/10208/commits/7f62120fdf26da4420d4267e4a5dd5cc2d366826), thanks for catching it.